instance/models/openedx_instance.py

```diff
@@ -213,17 +214,34 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
             )
         return backend_map, backend_conf
 
-    def set_active_vm_dns_records(self):
+    def set_active_vm_dns_records(self, deactivate_appserver=False):
         """
         Set DNS A records for all active app servers.
         """
         self.logger.info("Setting DNS records for active app servers...")
-        for i, appserver in enumerate(self.get_active_appservers(), 1):
-            ip_addr = appserver.server.public_ip
-            if ip_addr:
+        with cache.lock('appserver_dns_record_update_{}'.format(self.ref.instance_id)):
```

Comment:
@pkulkark, `self.ref.instance_id` points to `self.id`.

```suggestion
        with cache.lock('appserver_dns_record_update_{}'.format(self.id)):
```